department-of-veterans-affairs/va.gov-cms
# VACMS-15240 Stand up running next-build server per tugboat instance

## Description

Closes #15240. 

- Adds scripts to clone & install, build, and start the next-build repo as part of the cms tugboat environments. Mostly based on how content-build is included on tugboat currently through `.web`
- Creates apache config for tugboat in order to route requests with `next-*` to the running server

## Testing done
in the generated tugboat links below:
- visit the `next-*` prefix link and see that the 404 page is coming from next-build
- visit `/api/preview` and see a server error, because the config to communicate with drupal isn't there yet

- all existing generated tugboat links should resolve & behave as expected (`web-*`, `storybook-*`,  `cms-*`)

## Screenshots
![Screenshot 2023-10-18 at 4 04 57 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/11279744/4280dd8d-247f-46d7-afe7-7921aeb07eb6)
(this is what the 404 page from next.js looks like by default)


## QA steps
- all existing generated tugboat links should resolve & behave as expected (`web-*`, `storybook-*`,  `cms-*`)

Links:

https://pr15429-jnicnrrsvgrc1z8mpnqng14egyolhboj.ci.cms.va.gov/
https://cms-jnicnrrsvgrc1z8mpnqng14egyolhboj.ci.cms.va.gov/
should resolve to the drupal login screen

https://web-jnicnrrsvgrc1z8mpnqng14egyolhboj.ci.cms.va.gov/
should show the 403 page telling you to request a front end build. when that is requested & built, it should load the front end of va.gov.

https://storybook-jnicnrrsvgrc1z8mpnqng14egyolhboj.ci.cms.va.gov/
should show the 403 page telling you to check the logs for the storybook status. 
once it's built, it should resolve to the cms version of storybook

https://next-jnicnrrsvgrc1z8mpnqng14egyolhboj.ci.cms.va.gov/
before the server is running, it should return the 403 page like the others
once the server is running, it should resolve to a next.js 404 page 

https://next-jnicnrrsvgrc1z8mpnqng14egyolhboj.ci.cms.va.gov/api/preview should resolve to the 503 page, because the configuration between drupal & next-build isn't in place per env yet

### Definition of Done


### Select Team for PR review
